### PR TITLE
fix: remove publish to gh-pages from workflows

### PR DESCRIPTION
**Pull request by mas-who**
## Done

- Removed functionality to publish e2e reports to github pages in our workflows. The `gh-pages` branch history is causing bloat for the repo.

## QA

1. Run the LXD-UI:
    - On the demo server via the link posted by @webteam-app below. This is only available for PRs created by collaborators of the repo. Ask @mas-who or @edlerd for access.
    - With a local copy of this branch, [build and run as described in the docs](../CONTRIBUTING.md#setting-up-for-development).
2. Perform the following QA steps:
    - Ensure CI passes

### CODE REVIEW

File: .github/workflows/coverage.yaml [hunk 1]
```diff
@@ -110,60 +110,8 @@ jobs:
...
-            done
-        - name: Output Report URL as Worfklow Annotation
-          run: |
-            FULL_HTML_REPORT_URL=https://canonical.github.io/lxd-ui/coverage
-            echo "::notice title=Published Playwright Test Report::$FULL_HTML_REPORT_URL"
-
-
   tics-report:
     runs-on: ubuntu-latest
-    needs: publish-coverage-report
```
Comment (mas-who) [inline .github/workflows/coverage.yaml:166]: @edlerd not sure why `tics-report` depended on `publish-coverage-report`. From what I can see, just removing this line in this job should be fine, let me know if I missed something here

File: .github/workflows/coverage.yaml
Comment (edlerd) [inline .github/workflows/coverage.yaml:1]: We should keep the coverage report publishing, that is needed for the tics integration to fetch the latest coverage results.

**Review by edlerd — CHANGES_REQUESTED** [review]
Thx for bringing this up quickly. On review, I realized we can't completely remove the gh-pages branch, because we need to keep it for the coverage. But the payload there is much smaller than for the e2e reports, so should be fine to keep.

**Review by edlerd — APPROVED** [review]
LGTM, thanks for bringing this up so quickly!

**Pull request merged**